### PR TITLE
PENV-161: Config template generation

### PR DESCRIPTION
**Pull request by Vany**

### CODE REVIEW

File: configurator_test.go [hunk 1]
```diff
@@ -631,3 +632,28 @@ func Test_Load(t *testing.T) {
...
+func Test_TemplateTo(t *testing.T) {
+	x := &X{"1", "2", &Y{}, 3, 4, map[string]int{}, []int{}}
+	w := &bytes.Buffer{}
+	err := insconfig.NewYamlTemplater(x).TemplateTo(w)
+	require.NoError(t, err)
+	nx := X{}
+	require.NoError(t, yaml.Unmarshal(w.Bytes(), &nx))
+	require.NotNil(t, nx.E)
+
+}
```
[the remark this answers is not in the document]
Comment (Vany) [reply configurator_test.go:659]: nope, it's a template, it's empty

File: configurator.go [hunk 2]
```diff
@@ -336,3 +337,112 @@ func (i *insConfigurator) ToYaml(c interface{}) string {
...
+
+func NewYamlTemplater(obj interface{}) *YamlTemplater {
+	return &YamlTemplater{
+		Obj:   obj,
+		Level: -1,
+		Tag:   "",
+	}
+}
+
+func (m *YamlTemplater) TemplateTo(w io.Writer) error {
```
[the remark this answers is not in the document]
Comment (Vany) [reply configurator.go:364]: I thought about it, but we are creating exactly template

File: configurator_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MikhailPopov) [reply configurator_test.go:659]: I think Sergey means, it is worth to check expected YAML against the string(w.Bytes()). It is a good idea, isn't it?

File: configurator.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (SergeyRadist) [reply configurator.go:364]: Well, I can't see the result of TemplateTo, bacause you didn't added a test with expected plain config draft, so is it something like this ?
`port: {{ .Values.Port }}`

File: configurator_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MikhailPopov) [reply configurator_test.go:674]: I approved the changes after Vany added lines 661-672. JFYI @SergeyRadist )))

File: configurator.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Vany) [reply configurator.go:364]: and now ?

File: configurator.go [hunk 3]
```diff
@@ -336,3 +337,112 @@ func (i *insConfigurator) ToYaml(c interface{}) string {
...
+
+	if o, ok := m.Obj.(YamlTemplatable); ok {
+		return o.TemplateTo(w, m)
+	}
+	// TODO SOLVE me need work on (z *Z)TemplateTo()
```
Comment (megge-dream) [inline configurator.go:369]: is this todo still valid? if no, remove it, if yes, create task or bug for resolving it and mention task number (PENV-XXX) here

File: configurator_test.go [hunk 4]
```diff
@@ -631,3 +632,43 @@ func Test_Load(t *testing.T) {
...
+	A string `insconfig:"Adefault|large comment A with pipe='|'"`
+	B string `insconfig:"Bdefault|large comment B with pipe='|'"`
+	E *Y     `insconfig:"|---------------------------" yaml:"sacsacasc"`
+	C int
+	D uint8
+	G map[string]int
+	H []int
+}
+
+func Test_TemplateTo(t *testing.T) {
```
Comment (megge-dream) [inline configurator_test.go:650]: second test for error case would be great )

File: configurator.go (hunk 3 above)
Comment (Vany) [reply configurator.go:369]: no, it is not a task, it is a clue for task if it apears

Comment (megge-dream) [reply configurator.go:369]: then why TODO, if it's not a task?

**Pull request merged**